static/app/utils/useUser.tsx

```diff
@@ -0,0 +1,17 @@
+import configStore from 'sentry/stores/configStore';
+import {useLegacyStore} from 'sentry/stores/useLegacyStore';
+import type {User} from 'sentry/types';
+
+/**
+ * Returns the currently logged in user.
+ */
+export function useUser(): Readonly<User> {
+  // Intentional exception to accessing the deprecated field as we want to
+  // deter users from consuming the user differently than through the hook.
+  const {user} = useLegacyStore(configStore);
+  // @TODO: Return a readonly type as a mechanism to deter users from mutating the
+  // user directly. That said, this provides basic type safety and no runtime safety
+  // as there are still plenty of ways to mutate the user. The runtime safe way of
+  // enforcing this would be via Object.freeze.
+  return user;
```

Comment:
```suggestion
  // @TODO: Return a readonly value.
  // Currently we have a readonly type as a mechanism to deter users from mutating the
  // user directly. That said, this provides basic type safety and no runtime safety
  // as there are still plenty of ways to mutate the user. The runtime safe way of
  // enforcing this would be via Object.freeze.
  return user;
```

This comment would be helpful to future maintainers if we could explain why Object.freeze isn't in use yet.

Reply:
We cant enable it without centralizing any mutation logic first as it would risk blowing up in random places around the dashboard. I will add the comment, I thought it would be obvious

Reply:
I would not expect anyone to do that as I doubt anyone even reads these comments (at least I dont trust them to), but point taken.